core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -132,6 +134,23 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     assertTrue(meterCount(bytesOut) > initialBytesOut)
   }
 
+  @Test
+  def testBrokerTopicMetricsNoKeyCompactedTopicRecordsLogged(): Unit = {
```

Comment:
Seems we only have the test covering one out of the four metrics here, is this intentional?

Reply:
I've added assert statements for different metrics in:
- invalid magic number: https://github.com/apache/kafka/pull/7142/files#diff-0ad2dda25c093dbc8218289102f32997R85
- corrupted crc: https://github.com/apache/kafka/pull/7142/files#diff-3dc478d7151f94777a0f4fa188d2b2f4R118
- https://github.com/apache/kafka/pull/7142/files#diff-0ad2dda25c093dbc8218289102f32997R1209

Reply:
I have removed this test because https://github.com/apache/kafka/commit/600cc48fa56ca675421eaf383929e89a04b529f1 add a key to every message sent from `generateAndProduceMessages`. The test coverage for `NoKeyCompactedTopic` should be in `LogTest` now: https://github.com/apache/kafka/pull/7142/files#diff-37a823fbfd4840bf99b6395265723c5bR1826